### DISCUSSION TITLE
CPU Power

### DISCUSSION THREAD

**Discussion by rblancarte** [GerryDazoo/Slinger#67] [General]
This discussion is primarily to talk about what kind of systems you guys are installing Slinger on and how much CPU power you are using.

The reason I thought about this was because I had about a dozen Raspberry Pi 1s laying around and I was able to get this thing to work on my local network with no issues. I was able to stream to 2 different displays with no degrading of the signal. I might try 3 or more soon.

The main reason I want to get this to work is to stream our cable box to our place in Vermont. I really didn't want to dedicate a full up RPi4 to do this and using a 1st generation RPi would be perfect for this.

What other kinds of systems have you guys tried this on?

**Comment by f520k510** [comment]
I bought a raspberry pi 4 specifically for slinger purpose.  Then I soon found out I could install it on an oracle cloud VM.  It's free and runs 24/7.

**Reply by GerryDazoo** [reply]
One of my guys has this running on a Pi Nano. While steaming the overhead is very low, basically a packet in and a packet out for each simultaneous stream.
It's been run on every flavor or Windows and recently on Mac OS. Also many flavors of Linux, Raspberry Pi, Centos etc. I've got my server at home running on my Synology NAS box. We're close to getting it running on a iPad.
Basically any Windows platform to run the exe file if you want and any other platfrom where you can run Python3.

**Reply by DoogieHowitzerWMD** [reply]
> I bought a raspberry pi 4 specifically for slinger purpose. Then I soon found out I could install it on an oracle cloud VM. It's free and runs 24/7.

How does that Oracle cloud VM work exactly? I see where I can sign up for an account for the free tier, but I don't see much info on how it works exactly. I'm not sure if it's something that would be beyond my capabilities. Right now, I'm trying to run the server on a dedicated Raspberry Pi, but have been having some stability issues.

**Reply by DoogieHowitzerWMD** [reply]
> One of my guys has this running on a Pi Nano. While steaming the overhead is very low, basically a packet in and a packet out for each simultaneous stream. It's been run on every flavor or Windows and recently on Mac OS. Also many flavors of Linux, Raspberry Pi, Centos etc. I've got my server at home running on my Synology NAS box. We're close to getting it running on a iPad. Basically any Windows platform to run the exe file if you want and any other platfrom where you can run Python3.

I've been experimenting with running the server 24/7 on a dedicated headless Raspberry Pi (an old model 1B), but so far, I've suffered from a couple file corruption issues on the SD card. It seems the old Pi's like the one I'm using are worse with this issue than the newer ones.

To avoid/minimize this problem, there is a method of making the root file system read-only (enabling a Raspi-config setting called "Overlay File System"), where, from my understanding of it, a working copy of it, and any writes/changes made to it, are held in a temporary location (tmpfs). I think those are held in RAM, which is on short supply on my old Pi (256 Mb). If the Sling server needs that filesystem, continuous running of the server seems like it would eventually max out the available RAM and grind the Pi to a halt.

I might need to think about another solution for constantly running the server if there's not a way to get that sorted out.

**Comment by GerryDazoo** [comment]
Please note: You don't have to have the server running in the same location as the slingbox. You can configure the server to connect to your SB at your remote location. All you need to do is add a port forwarding rule to the router in Vermont to allow TCP connections to your slingbox.

**Reply by GerryDazoo** [reply]
Sorry, I misread the post. The port forwarding rule isn't on the router in vermont, it's on the router where the slingbox is located. You would then run the server at your place in Vermont. Some people have the server setup on their laptop and can use their slingbox at home from anywhere they may travel. There is no mandatory reason to have a dedicated sever running at home.
